docs/developers-guide/cadl-docs/README.md

```diff
@@ -2,158 +2,321 @@
+
+Elm:
+``` elm
+type alias AccountGroup = 
+    Char
 ```
-alias Char = string
+Cadl:
+```cadl
+alias AccountGroup = string;
 ```
 
 ### Advanced Types
 ##### [Decimal](https://package.elm-lang.org/packages/finos/morphir-elm/18.1.0/Morphir-SDK-Decimal)
 The `decimal` type in morphir defines a floating point number with accurate precision. This type is not supported directly in CADL and an alternative mapping is the `string` type.
+
+Elm:
+```elm
+import Morphir.SDK.Decimal exposing (Decimal)
+
+type alias Price = 
+    Decimal
 ```
-alias Decimal = string
+Cadl:
+```cadl
+alias Price = string
 ```
 ##### [LocalDate](https://package.elm-lang.org/packages/finos/morphir-elm/18.1.0/Morphir-SDK-LocalDate)
-The `localDate` type in morphir defines as a gregorian date with no timezone information. This type maps directly to `plainDate` type in CADL. 
+The `localDate` type in morphir defines as a gregorian date with no timezone information. This type maps directly to `plainDate` type in CADL.
+
+Elm:
+```elm
+import Morphir.SDK.LocalDate
+
+type alias localDate = 
+    Date
 ```
-alias dateOfBirth = plainDate
+Cadl:
+```cadl
+alias localDate = plainDate;
 ```
 ##### [LocalTime](https://package.elm-lang.org/packages/finos/morphir-elm/18.1.0/Morphir-SDK-LocalTime)
-The `localTime` type in morphir defines as basic time without a timezone and its equivalent mapping in CADL is `plainTime` type. 
+The `localTime` type in morphir defines as basic time without a timezone and its equivalent mapping in CADL is `plainTime` type.
+
+Elm:
+```elm
+import Morphir.SDK.LocalTime
+
+type alias LocalTime =
+    Posix
 ```
-alias LocaTime = plainTime
+Cadl:
+```cadl
+alias LocaTime = plainDate;
 ```
+
 ##### [Month](https://package.elm-lang.org/packages/finos/morphir-elm/18.1.0/Morphir-SDK-Month)
-The morphir type `month` derived from a localDate was purposefully created to aid in business modeling. This concept of `month` type does not 
+The morphir type `month` derived from a localDate was purposefully created to aid in business modeling. This concept of `month` type does not
 exist in CADL and the alternative mapping is the `string` type.
+
+Elm:
+```elm
+import Morphir.SDK.Month
+
+type alias purchaseMonth =
+    Month
 ```
-alias Month = string
+Cadl:
+```cadl
+alias purchaseMonth = string;
 ```
+
 ##### [Optional Values(Maybe)](https://package.elm-lang.org/packages/elm/core/latest/Maybe)
-The `maybe` type in morphir represents values that or may not exist. This is directly supported in CADL through `optional fields` of models or as alias.
-```
-Examples : 
-    1. As an optional field of a model using the `?:` syntax. 
-        model Foo {
-            foo ?: string
-        }
-        
-    2. As an alias with of type T and null.
-        alias mayBe<T> = T | null
-```
+The `maybe` type in morphir represents values that may or may not exist. This is directly supported in CADL through:
+
+1. `optional field` in a model using the `?:` syntax.
```

Comment:
You should explain that this will be done only when the optional value is a field inside a record. In any other case the second mapping will be used. Maybe you should also mention that in case of nested `Maybe`s the top level maybe will be removed and turned into an optional field.